File: MapReduce/Surbhi_Batra_task1.scala
import org.apache.spark
import org.apache.spark.sql.SQLContext
import org.apache.spark.sql.SparkSession
import org.apache.spark.{SparkConf, SparkContext}
import org.apache.spark.sql.functions.collect_list
import org.apache.spark.sql.functions._

object Task1 {
  def main(args: Array[String]): Unit = {
    val infile = args(0)
    val outfile = args(1)
    //    val conf = new SparkConf().setAppName("Task1").setMaster("local[2]")
    //    val sc = new SparkContext(conf)
    //    println(sc)

    val spark = org.apache.spark.sql.SparkSession.builder
      .master("local")
      .appName("Spark CSV Reader")
      .getOrCreate;

    //    val sqlContext = new org.apache.spark.sql.SQLContext(sc)
    //    import sqlContext.implicits._
    //    val file = sc.textFile(infile)
    //
    //    val df = file.map(line => line.split(",")).
    //      filter(lines => lines(0)!= "userId").
    //      map(row => (row(0), row(1), row(2), row(4))).
    //      toDF("userId", "movieId","rating","timestamp")

    val df = spark.read
      .format("csv")
      .option("header", "true") //reading the headers
      .option("mode", "DROPMALFORMED")
      .option("inferSchema", "True")
      .load(infile)
    //      .load("ml-latest-small/ratings.csv")

    val result = df.groupBy("movieId").agg(mean("rating").alias("rating_avg"))
    result.orderBy("movieId")
      .coalesce(1)
      .write.format("com.databricks.spark.csv")
      .option("header", "true")
      .save(outfile)
    //      .save("scala_task1_small.csv")
  }
}

File: MapReduce/Surbhi_Batra_task1.py
import pyspark as spark

from pyspark.sql import SparkSession

from pyspark.sql import functions as F

from pyspark.sql.types import DoubleType

import sys
infile = sys.argv[1]
outfile = sys.argv[2]

spark = SparkSession \
		.builder \
		.appName("Task1_try1") \
		.config("spark.some.connfig.option","some-value") \
		.getOrCreate()

# df = spark.read.csv("/home/surbhi/Documents/2018Summer/DM/Assignment_01/ml-latest-small/ratings.csv", header = True)
df = spark.read.csv(infile, header = True, inferSchema= True)
# df = df.withColumn("movieId", df["movieId"].cast(DoubleType()))
df = df.groupBy("movieId").agg(F.avg("rating").alias('rating_avg'))
df = df.orderBy("movieId")
df.repartition(1).write.format("com.databricks.spark.csv").option("header", "true").save(outfile)
df.show()

#$SPARK_HOME/bin/spark-submit solution/surbhi_batra_task1.py solution/ml-latest-small/ratings.csv outputFiles/surbhi_batra_result_task1_small.csv
#$SPARK_HOME/bin/spark-submit solution/surbhi_batra_task1.py solution/ml-20m/ratings.csv outputFiles/surbhi_batra_result_task1_big.csv
